packages/perseus-editor/src/components/graph-locked-figures/locked-function-settings.tsx

```diff
@@ -0,0 +1,246 @@
+/**
+ * LockedFunctionSettings is a component that allows the user to edit the
+ * settings of specifically a locked function on the graph.
+ *
+ * Used in the interactive graph editor's locked figures section.
+ */
+import {UniqueIDProvider, View} from "@khanacademy/wonder-blocks-core";
+import {OptionItem, SingleSelect} from "@khanacademy/wonder-blocks-dropdown";
+import {TextField} from "@khanacademy/wonder-blocks-form";
+import {Strut} from "@khanacademy/wonder-blocks-layout";
+import {spacing} from "@khanacademy/wonder-blocks-tokens";
+import {LabelLarge, LabelMedium} from "@khanacademy/wonder-blocks-typography";
+import {StyleSheet} from "aphrodite";
+import * as React from "react";
+import {useEffect, useState} from "react";
+
+import PerseusEditorAccordion from "../perseus-editor-accordion";
+
+import ColorSelect from "./color-select";
+import LineStrokeSelect from "./line-stroke-select";
+import LineSwatch from "./line-swatch";
+import LockedFigureSettingsActions from "./locked-figure-settings-actions";
+
+import type {LockedFigureSettingsCommonProps} from "./locked-figure-settings";
+import type {LockedFunctionType} from "@khanacademy/perseus";
+import type {Interval} from "mafs";
+
+export type Props = LockedFunctionType &
+    LockedFigureSettingsCommonProps & {
+        /**
+         * Called when the props (points, color, etc.) are updated.
+         */
+        onChangeProps: (newProps: Partial<LockedFunctionType>) => void;
+    };
+
+const LockedFunctionSettings = (props: Props) => {
+    const {
+        color: lineColor,
+        strokeStyle,
+        equation,
+        directionalAxis,
+        domain,
+        onChangeProps,
+        onMove,
+        onRemove,
+    } = props;
+    const equationPrefix = directionalAxis === "x" ? "y=" : "x=";
+    const lineLabel = `Function (${equationPrefix}${equation})`;
+    const domainLimits = domain ? [...domain] : [-Infinity, Infinity];
+
+    // Tracking the string value of domain values to handle interim state of
+    //     entering in a negative value - used when specifying value of input field
+    const [domainEntries, setDomainEntries] = useState([
+        domain && domain[0] !== -Infinity ? domain[0].toString() : "",
+        domain && domain[1] !== Infinity ? domain[1].toString() : "",
+    ]);
+
+    useEffect(() => {
```

Comment:
It sounds like this useEffect is here to handle the domain changing to/from infinity? Is that right?

Could you add a commenting explaining why this useEffect is necessary?

Reply:
Comment added.